components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -3411,6 +3439,37 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, &
                       - exp( max(-100.0_RKIND, &
                                  -layerThickness(kmin, iCell)/ &
                            config_flux_attenuation_coefficient_runoff))
+         if (trim(config_subglacial_runoff_mode) == 'data') then
+             if (config_use_sgr_opt_kpp == 1) then
+                if ( trim(config_sgr_flux_vertical_location) == 'top' ) then
```

Comment:
Similar to diagnostics, would it make sense to add and `else` this this and raise an error if `config_sgr_flux_vertical_location` has an unexpected value?  Otherwise, I worry that a typo could run fine but have weird and hard-to-explain behavior.

Reply:
Done